src/views/compute/wizard/ComputeOfferingSelection.vue

```diff
@@ -0,0 +1,164 @@
+      <span slot="ramTitle"><a-icon type="bulb" /> {{ $t('memory') }}</span>
+    </a-table>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'ComputeOfferingSelection',
+  props: {
+    computeItems: {
+      type: Array,
+      default: () => []
+    },
+    value: {
+      type: String,
+      default: ''
+    },
+    loading: {
+      type: Boolean,
+      default: false
+    },
+    preFillContent: {
+      type: Object,
+      default: () => {}
+    }
+  },
+  data () {
+    return {
+      filter: '',
+      columns: [
+        {
+          dataIndex: 'name',
+          title: this.$t('serviceOfferingId'),
+          width: '40%'
+        },
+        {
+          dataIndex: 'cpu',
+          slots: { title: 'cpuTitle' },
+          width: '30%'
+        },
+        {
+          dataIndex: 'ram',
+          slots: { title: 'ramTitle' },
+          width: '30%'
+        }
+      ],
+      selectedRowKeys: []
+    }
+  },
+  computed: {
+    options () {
+      return {
+        page: 1,
+        pageSize: 10,
+        keyword: ''
+      }
+    },
+    tableSource () {
+      return this.computeItems.map((item) => {
+        var cpuNumberValue = item.cpunumber + ''
+        var cpuSpeedValue = (item.cpuspeed !== null && item.cpuspeed !== undefined && item.cpuspeed > 0) ? parseFloat(item.cpuspeed / 1000.0).toFixed(2) + '' : ''
+        var ramValue = item.memory + ''
+        if (item.iscustomized === true) {
+          cpuNumberValue = ''
+          ramValue = ''
+          if ('serviceofferingdetails' in item &&
+            'mincpunumber' in item.serviceofferingdetails &&
+            'maxcpunumber' in item.serviceofferingdetails) {
+            cpuNumberValue = item.serviceofferingdetails.mincpunumber + '-' + item.serviceofferingdetails.maxcpunumber
+          }
+          if ('serviceofferingdetails' in item &&
+            'minmemory' in item.serviceofferingdetails &&
+            'maxmemory' in item.serviceofferingdetails) {
+            ramValue = item.serviceofferingdetails.minmemory + '-' + item.serviceofferingdetails.maxmemory
+          }
+        }
+        return {
+          key: item.id,
+          name: item.name,
+          cpu: cpuNumberValue.length > 0 ? `${cpuNumberValue} CPU x ${cpuSpeedValue} Ghz` : '',
+          ram: ramValue.length > 0 ? `${ramValue} MB` : ''
+        }
+      })
+    },
+    rowSelection () {
+      return {
+        type: 'radio',
+        selectedRowKeys: this.selectedRowKeys,
+        onChange: this.onSelectRow
+      }
+    }
+  },
+  watch: {
+    value (newValue, oldValue) {
+      if (newValue && newValue !== oldValue) {
+        this.selectedRowKeys = [newValue]
+      }
+    },
+    loading () {
+      if (!this.loading && this.preFillContent.computeofferingid) {
+        this.selectedRowKeys = [this.preFillContent.computeofferingid]
+        this.$emit('select-compute-item', this.preFillContent.computeofferingid)
+      }
```

Comment:
A minor improvement here can be selecting first element on list refresh when there is no `this.preFillContent.computeofferingid` and previously selected offering is not available in the list. Else deployVM API is fired with previous value without even selecting anything in the new list.
Since offerings and network can depend on selected same behaviour could be seen with other parameters(disk, network) as well